jobrunner/executors/local.py

```diff
@@ -241,11 +241,7 @@ def finalize_job(job):
     container_metadata = get_container_metadata(job)
     outputs, unmatched_patterns = find_matching_outputs(job)
     exit_code = container_metadata["State"]["ExitCode"]
-    message = None
-    if exit_code == 137:
-        # 137 = 128+9, which means was killed by signal 9, SIGKILL
-        # This usually happens because of OOM killer, or else manually
-        message = "likely means it ran out of memory"
+    message = config.EXIT_CODES.get(exit_code)
```

Comment:
I think we should treat these exit codes as being specific to database jobs. So if, say, a random R job happens to exit 3 we don't report it as a database error.

Fortunately, I think this should just be a case of checking the `job.allow_database_access` flag. But it does make things a bit more awkward in that we need separate handling for the 137 code. Maybe we just need two separate error code dicts, one for database jobs and one for the rest? Not sure what the cleanest approach is here.